x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -436,45 +434,28 @@ The Kibana Connector in use may need to be reconfigured with an updated Amazon B
   }
 
   /**
-   * Sends a request to the Bedrock API to perform a conversation action.
-   * @param input - The parameters for the conversation action.
+   * Sends a request via the BedrockRuntimeClient to perform a conversation action.
+   * @param params - The parameters for the conversation action.
+   * @param params.signal - The signal to cancel the request.
+   * @param params.command - The command class to be sent to the API. (ConverseCommand | ConverseStreamCommand)
    * @param connectorUsageCollector - The usage collector for the connector.
    * @returns A promise that resolves to the response of the conversation action.
    */
-  public async converse(
-    { signal, ...converseApiInput }: ConverseActionParams,
+  public async bedrockClientSend(
+    { signal, command }: ConverseActionParams,
     connectorUsageCollector: ConnectorUsageCollector
   ): Promise<ConverseActionResponse> {
-    const res = await this.runApi(
-      {
-        body: JSON.stringify(converseApiInput),
-        raw: true,
-        apiType: 'converse',
-        signal,
-      },
-      connectorUsageCollector
-    );
-    return res;
-  }
+    connectorUsageCollector.addRequestBodyBytes(undefined, command);
+    const res = await this.bedrockClient.send(command, {
+      abortSignal: signal,
+    });
 
-  /**
-   * Sends a request to the Bedrock API to perform a streaming conversation action.
-   * @param input - The parameters for the streaming conversation action.
-   * @param connectorUsageCollector - The usage collector for the connector.
-   * @returns A promise that resolves to the streaming response of the conversation action.
-   */
-  public async converseStream(
-    { signal, ...converseApiInput }: ConverseActionParams,
-    connectorUsageCollector: ConnectorUsageCollector
-  ): Promise<IncomingMessage> {
-    const res = await this.streamApi(
-      {
-        body: JSON.stringify(converseApiInput),
-        apiType: 'converse',
-        signal,
-      },
-      connectorUsageCollector
-    );
+    if ('stream' in res) {
+      const resultStream = res.stream as SmithyMessageDecoderStream<unknown>;
+      // splits the stream in two, [stream = consumer, tokenStream = token tracking]
+      const [stream, tokenStream] = tee(resultStream);
+      return { ...res, stream, tokenStream };
+    }
```

Comment:
@patrykkopycinski I wrote my own `tee` function to split the iterator into two streams which can be independently read from at different speeds. 🎉